screens/HomeScreen.tsx

```diff
@@ -9,10 +14,27 @@ export const screenStyles = StyleSheet.create({
   },
 });
 
-const HomeScreen = () => {
+const HomeScreen = ({ navigation }) => {
+  if (auth.currentUser == null) {
```

Comment:
Expanding on @phoebeli23's  feedback: This isn't correctly handling null users - the Firebase console is probably updating between between this null check and the actual return statement, which means this check is getting overridden. Linking resource from same medium article. They wrap this error handling in useEffect, which always gets called when component first mounts. I'm not 100% sure if it'll solve your issue, but worth a try! The explanation is pretty good! 

[Snippet.](https://medium.com/swlh/expo-firebase-authentication-cloud-firestore-using-async-await-react-hooks-700920ad4b2b) Cmd+F 'Step 9: Dashboard and Sign Out' 

![Screen Shot 2021-11-03 at 12 37 08 AM](https://user-images.githubusercontent.com/63089231/140023536-afb1cb7e-f936-4f16-838b-90529fb0b241.png)